### PR TITLE
fix: cypress-schematic add exception for nguniversal ssr dev server

### PR DESCRIPTION
**Pull request by conradkirschner**
- Closes https://github.com/cypress-io/cypress/issues/23345

### User facing changelog
Allow optional watch option for angular universal dev server 

### How has the user experience changed?
It will now work with angular universal dev server 

### PR Tasks

- [ na] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (ZachJW34) [inline npm/cypress-schematic/src/builders/cypress/index.ts:106]: The [nrwl executor linked](https://github.com/nrwl/nx/blob/f930117ed6ab13dccc40725c7e9551be081cc83d/packages/cypress/src/executors/cypress/cypress.impl.ts#L163) checks the devServerTarget options, not the string itself. Most `devServerTarget` strings don't contain `watch` in the name, so this would disable watch for projects that do support it. We should use a similar API that Nx uses (`readTargetOptions`) to get the keys if Angular offers one.

Otherwise, an explicit hardcoding might be a quick fix for nguniversal if we can detect the executor/builder from the devServerTarget

**Review by ZachJW34 — CHANGES_REQUESTED** [review]

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (conradkirschner) [reply npm/cypress-schematic/src/builders/cypress/index.ts]: Hey  @ZachJW34 , 
first of all thanks for fast response. I tried to fix it, but I am super unsure about the changes. 
I tried to readout the options like nrwl, but this will bloat the builder a lot - therefore I went for option 2. 
Only add watch property if not universal ssr dev server as selected target. 

I also added a notification for the dev as hint that watch property will be added.

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (ZachJW34) [inline npm/cypress-schematic/src/builders/cypress/index.ts:106]: Rather than read the config, can you try utilizing `context.getBuilderNameForTarget(targetFromTargetString(devServerTarget))`? That should give you the executor you need.

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (lmiller1990) [inline npm/cypress-schematic/src/builders/cypress/index.ts:114]: Instead of making `watch` optional should we just do

```js
coinst overrides = {
  watch: devServerConfig.builder !== '@nguniversal/builders:ssr-dev-server' ? watch : false
}
```

Just a nit/optional update, if we get more things to ignore in the future can do

```ts
const ignoreWatchBuilders = ["@nguniversal/builders:ssr-dev-server"]

overrides = {
  watch: [ignoreWatchBuilders].some(({ builder }) => builder === devServerConfig.builder) ?  false : watch
```

**Review by lmiller1990 — APPROVED** [review]
Just a nit, not blocking - will approve so once the final update is made we can merge.

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (conradkirschner) [reply npm/cypress-schematic/src/builders/cypress/index.ts:114]: watch has to be not there, otherwise the json schema validation will fail - also it doesnt allow additional attributes. 

watch has to be optional

**Review by lmiller1990 — APPROVED** [review]

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (conradkirschner) [reply npm/cypress-schematic/src/builders/cypress/index.ts]: Ah, yes I was confused because I thought these were the Dev Server Props - not the Props that cypress passing to the dev server ... 

Renamed it for more clearification. (prefixed with cypress)

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (ZachJW34) [inline npm/cypress-schematic/src/builders/cypress/index.ts:22]: Not sure why this is typed as `any`, can you try changing it to `BuilderContext` so that the `startDevServer` function has proper typings?

File: npm/cypress-schematic/src/builders/cypress/index.ts
Comment (ZachJW34) [inline npm/cypress-schematic/src/builders/cypress/index.ts:50]:
```suggestion
        ? startDevServer({ devServerTarget: options.devServerTarget, watch: options.watch, context }).pipe(
          map((devServerBaseUrl: string) => options.baseUrl || devServerBaseUrl),
```

Should be able to drop `forkJoin` now that we've refactored the `startDevServer` to return a single observable

**Review by ZachJW34 — COMMENTED** [review]
One last tweak then should be good to go!

**Review by ZachJW34 — APPROVED** [review]
LGTM!!!

**Comment by ZachJW34** [comment]
@conradkirschner I changed to the base branch to `master` as that is our branch that will automatically release our npm packages come merge

**Review by lmiller1990 — APPROVED** [review]

**Comment by conradkirschner** [comment]
@ZachJW34  It got merged but not published. [workflow](https://github.com/cypress-io/cypress/actions/runs/2871983599
Looks like bot doesnt have permission for merge into dev and it wasnt published on npm. 

@ZachJW34 @lmiller1990

**Comment by ZachJW34** [comment]
@conradkirschner that failure is (unfortunately) expected, but there were some tests that failed which look like they don't pertain to your changes. I'm rerunning CI, the [npm-release](https://app.circleci.com/pipelines/github/cypress-io/cypress/42032/workflows/8b4fca31-641f-40bf-a667-3ebe25ea8f0b) job is the job to keep an eye on

**Comment by ZachJW34** [comment]
[@cypress/schematic@2.0.3](https://www.npmjs.com/package/@cypress/schematic) has been published!

**Pull request merged**
